Bottom: a7de1dbd0ce6d2184f8385aac8fd0a9d5ca64f67
Top:    49eedc84777de0854ece2c0a1d7bbc239a2c09c3
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 12:19:58 -0500

Use install to copy key and crt

Install will set mode and permissions in one step.


---
diff --git a/docker/docker-exim4/docker-entrypoint.sh b/docker/docker-exim4/docker-entrypoint.sh
index 3c3e0a26..77a575ee 100755
--- a/docker/docker-exim4/docker-entrypoint.sh
+++ b/docker/docker-exim4/docker-entrypoint.sh
@@ -25,17 +25,11 @@ case ${1} in
 
         # copy certificate
         [ -e /etc/ssl/private/exim.crt ] \
-            && rm -f ${CONFIG_PATH}/exim.crt \
-            && cp $(realpath /etc/ssl/private/exim.crt) ${CONFIG_PATH}/exim.crt \
-            && chmod 640 ${CONFIG_PATH}/exim.crt \
-            && chown root:Debian-exim ${CONFIG_PATH}/exim.crt
+            && install -oroot -gDebian-exim -m640 $(realpath /etc/ssl/private/exim.crt) ${CONFIG_PATH}/exim.crt
 
         # copy key
         [ -e /etc/ssl/private/exim.key ] \
-            && rm -f ${CONFIG_PATH}/exim.key \
-            && cp $(realpath /etc/ssl/private/exim.key) ${CONFIG_PATH}/exim.key \
-            && chmod 640 ${CONFIG_PATH}/exim.key \
-            && chown root:Debian-exim ${CONFIG_PATH}/exim.key
+            && install -oroot -gDebian-exim -m640 $(realpath /etc/ssl/private/exim.key) ${CONFIG_PATH}/exim.key
 
         # run the debian utility to update/generate the configuration
         $(which update-exim4.conf) --verbose
